internal/pkg/api/server/buffer/dependencies/dependencies.go

```diff
@@ -283,6 +285,16 @@ func (v *forServer) EtcdClient(ctx context.Context) (*etcd.Client, error) {
 	})
 }
 
+func (v *forServer) ConfigStore(ctx context.Context) (*ConfigStore, error) {
+	return v.configStore.InitAndGet(func() (*ConfigStore, error) {
+		_, err := v.EtcdClient(ctx)
+		if err != nil {
+			return nil, err
+		}
+		return NewConfigStore(v), nil
```

Comment:
Load `etcdClient` once, in `NewConfigStore`.

Reply:
A server bez `etcd` nenastartuje, vratiti `err` tu:
https://github.com/keboola/keboola-as-code/blob/db3935b2b6d111fd6dff5e6d75a15fc75d54cf2d/internal/pkg/api/server/buffer/dependencies/dependencies.go#L156-L160

Reply:
Este to mozem napisat konkretnejsie (@jprochazka):
- Teda `EtcdClient` je vzdy dostupny a tento interface nevracia chybu: `EtcdClient(ctx context.Context) (*etcd.Client, error)`: 
  - https://github.com/keboola/keboola-as-code/blob/db3935b2b6d111fd6dff5e6d75a15fc75d54cf2d/internal/pkg/api/server/buffer/dependencies/dependencies.go#L69 (podobne aj interface v ConfigStore).
- Odtial mozes odstranit `Lazy`: https://github.com/keboola/keboola-as-code/blob/db3935b2b6d111fd6dff5e6d75a15fc75d54cf2d/internal/pkg/api/server/buffer/dependencies/dependencies.go#L94
- Tento kod a dalej: https://github.com/keboola/keboola-as-code/blob/db3935b2b6d111fd6dff5e6d75a15fc75d54cf2d/internal/pkg/api/server/buffer/dependencies/dependencies.go#L210-L232
    - Mozes presunut do nejakej novej private funkcie `newEtcdClient`.
- A tu pouzijes `newEtcdClient` funkciu, ak `err` tak ju vratis, inak nastavis hodnotu `d.etcdClient`
  - https://github.com/keboola/keboola-as-code/blob/db3935b2b6d111fd6dff5e6d75a15fc75d54cf2d/internal/pkg/api/server/buffer/dependencies/dependencies.go#L156-L159
- Tym mas zabezpecene:
  - Server nenastartuje bez etcd spojenia.
  - Etcd klient je vzdy dostupny, nemusis kontrolovat chybu, ked ho z `d` ziskavas.
  - Ak by nastal nejaky sietovy problem, tak sa pri konkretnej `etcd` operacii klient snazi znova pripojit - je tam retry.